flink-cdc-common/src/main/java/org/apache/flink/cdc/common/utils/SchemaUtils.java

```diff
@@ -101,6 +101,9 @@ public static Schema applySchemaChangeEvent(Schema schema, SchemaChangeEvent eve
     private static Schema applyAddColumnEvent(AddColumnEvent event, Schema oldSchema) {
         LinkedList<Column> columns = new LinkedList<>(oldSchema.getColumns());
         for (AddColumnEvent.ColumnWithPosition columnWithPosition : event.getAddedColumns()) {
+            if (oldSchema.getColumnNames().contains(columnWithPosition.getAddColumn().getName())) {
+                continue;
+            }
```

Comment:
Seems we don't need this check after `SchemaUtils#isSchemaChangeEventRedundant` has guarded us from duplicately applying non-idempotent schema change events.